fc/parsing/actions.py

```diff
@@ -882,6 +878,150 @@ def _expr(self):
+        self.model_variable = None
+        self.transitive_variables = set()
+
+    @property
+    def long_name(self):
+        if self._aliases:
+            return self.name + '(aka ' + ', '.join(self._aliases) + ')'
+        else:
+            return self.name
+
+    def update(self, name=None, input=False, output=False, output_category=False, optional=False, local=False,
+               units=None, initial_value=None, default_expr=None, equation=None, model_variable=None,
+               transitive_variables=None):
+        """
+        Merges new information into this :class:`ProtocolVariable`, raises a ProtocolError if new information is
+        incompatible with what is already stored.
+        """
+        # Store name
+        if name and name not in self._aliases:
+            self._aliases.append(name)
+
+        # Merge type information
+        self.input = self.input or input
+        self.output = self.output or output
+        self.output_category = self.output_category or output_category
+        self.optional = self.optional or optional
+        self.local = self.local or local
+
+        # Check type information isn't conflicting
+        assert not (self.local and (self.input or self.output)), f'{self.long_name} is local AND input/output'
+
+        # Merge units
+        if self.units is None:
+            self.units = units
+        elif units is not None and units != self.units:
+            raise ProtocolError(f'Inconsistent units specified for {self.long_name} ({self.units} and {units}).')
+
+        # Merge initial value
+        if self.initial_value is None:
+            self.initial_value = initial_value
+        elif initial_value is not None and initial_value != self.initial_value:
+            raise ProtocolError(f'Multiple initial values specified for {self.long_name} ({self.initial_value} and'
+                                ' {initial_value}).')
+
+        # Merge default expression
+        if self.default_expr is None:
+            self.default_expr = default_expr
+        elif default_expr is not None and default_expr != self.default_expr:
+            raise ProtocolError(f'Multiple default expressions specified for {self.long_name} ({self.default_expr} and'
+                                ' {default_expr}).')
+
+        # Merge equation
+        if self.equation is None:
+            self.equation = equation
+        elif equation is not None and equation != self.equation:
+            raise ProtocolError(f'Multiple equations specified for {self.long_name}.')
+
+        # Set model variable
+        if model_variable is not None:
+            if self.model_variable is None:
+                self.model_variable = model_variable
+            else:
+                assert model_variable == self.model_variable, 'ProtocolVariable representing multiple model variables'
+
+        # Add transitive variables
+        if transitive_variables:
+            self.transitive_variables.update(transitive_variables)
+
```

Comment:
We should probably `assert not self.transitive_variables or self.model_variable is None` here since it shouldn't be possible for one reference to be both a single variable and a vector.